src/main/scala/vsys/Application.scala

```diff
@@ -46,7 +46,7 @@ class Application(val actorSystem: ActorSystem, val settings: VsysSettings) exte
   private val db = openDB(settings.dataDirectory)
 
   private val checkpointService = new CheckpointServiceImpl(db, settings.checkpointsSettings)
-  private val (history, _, stateReader, blockchainUpdater) = StorageFactory(db, settings.blockchainSettings)
+  private val (history, _, stateReader, blockchainUpdater) = StorageFactory(db, settings.blockchainSettings, settings.eventSettings, actorSystem)
```

Comment:
May build a independent singleton out of `StrorageFactory` with  `settings.eventSettings` and `actorSystem`
this module is not a part of storage system

Reply:
we can form an `EventTrigger` with `EventWriterActor` and `EventSettings` in application.scala and pass this trigger to `StorageFactory`